src/pyodide-py/pyodide/_base.py

```diff
@@ -49,55 +49,38 @@ class CodeRunner:
 
     Examples
     --------
-    >>> CodeRunner("1+1", {}).run()
+    >>> CodeRunner().run("1+1")
     2
-    >>> CodeRunner("1+1;", {}).run()
+    >>> CodeRunner().run("1+1;")
     """
 
-    def __init__(self, code: str, ns: Dict[str, Any]):
+    def __init__(self, ns: Dict[str, Any] = {}):
         """
         Constructor.
 
         Parameters
         ----------
-        code
-           the Python code to run.
         ns
            `locals()` or `globals()` context where to execute code.
         """
         # handle mis-indented input from multi-line strings
-        code = dedent(code)
 
-        self.code = code
         self.ns = ns
         self.filename = "<exec>"
-        self.mod = None
-        self.last_expr = None
-
-        mod = ast.parse(code)
-        if len(mod.body) == 0:
-            return
-
-        # we extract last expression
-        last_expr = None
-        if isinstance(mod.body[-1], (ast.Expr, ast.Await)) and not self.quiet():
-            last_expr = ast.Expression(mod.body.pop().value)  # type: ignore
-        self.mod = mod
-        self.last_expr = last_expr
 
-    def quiet(self) -> bool:
+    def quiet(self, code: str) -> bool:
```

Comment:
Can probably even be a `@staticmethod` ?

Reply:
In the future, the `quiet` behavior can be impacted by options from the constructor (see #1056).